app/lib/http/movie/movie.service.ts

```diff
@@ -0,0 +1,37 @@
+import { GetMethodParams, get } from "../request";
+
+import { ExtractResponseType } from "@/app/types";
+
+import { paths } from "@/app/tmdb_schema";
+
+const BASE_PATH_MOVIE = "movie";
+const PATH_MOVIES = {
+  POPULAR: `${BASE_PATH_MOVIE}/popular`,
+};
+
+export type FetchPopularMoviesResponse =
+  paths["/3/movie/popular"]["get"]["responses"]["200"]["content"]["application/json"];
+
+export type PopularMovie = ExtractResponseType<FetchPopularMoviesResponse["results"]>;
+
+export const fetchPopularMovies = async (
+  params?: GetMethodParams["params"],
+): Promise<FetchPopularMoviesResponse> => {
+  try {
+    const response = await get({
+      path: PATH_MOVIES.POPULAR,
+      params: {
+        ...params,
+      },
+    });
+
+    if (response.ok === false) {
+      throw response;
+    }
+
+    return await response.json();
+  } catch (error) {
+    console.log("error: ", error);
+    throw error;
```

Comment:
You are throwing in this catch block but there is no consumer that handles it so I think the app eventually breaks when the response is anything besides `200`. I'd recommend you set up general error handling logic here rather than throwing it so that no consumers need to focus on error handling and they won't have a risk of crash error.

The founder of Builder.io, Steve have written a [great article about this](https://www.builder.io/blog/safe-data-fetching) so it'd be a nice read for you on this topic 👍

Reply:
Sure, I created the issue and will tackle on #7